kong/acl_service_test.go

```diff
@@ -137,6 +137,50 @@ func TestACLGroupGet(T *testing.T) {
 	require.NoError(client.Consumers.Delete(defaultCtx, consumer.ID))
 }
 
+func TestACLGroupGetByID(T *testing.T) {
+	RunWhenDBMode(T, "postgres")
+
+	assert := assert.New(T)
+	require := require.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(err)
+	assert.NotNil(client)
+
+	uuid := uuid.NewString()
+	acl := &ACLGroup{
+		ID:    String(uuid),
+		Group: String("my-group"),
+	}
+
+	// consumer for the ACLGroup
+	consumer := &Consumer{
+		Username: String("foo"),
+	}
+
+	consumer, err = client.Consumers.Create(defaultCtx, consumer)
+	require.NoError(err)
+	require.NotNil(consumer)
+
+	createdACL, err := client.ACLs.Create(defaultCtx, consumer.ID, acl)
+	require.NoError(err)
+	assert.NotNil(createdACL)
+
+	aclGroup, err := client.ACLs.GetByID(defaultCtx, acl.ID)
+	require.NoError(err)
+	assert.Equal("my-group", *aclGroup.Group)
+
+	aclGroup, err = client.ACLs.GetByID(defaultCtx, String("does-not-exist"))
+	assert.Nil(aclGroup)
+	require.Error(err)
+
+	aclGroup, err = client.ACLs.GetByID(defaultCtx, String(""))
+	assert.Nil(aclGroup)
+	require.Error(err)
+
+	require.NoError(client.Consumers.Delete(defaultCtx, consumer.ID))
```

Comment:
Use `t.Cleanup()` for such operations. That makes the intent clear.
We should also cleanup the other entity that you created here - the cred.
Refer: https://github.com/Kong/go-kong/blob/main/kong/vault_service_test.go#L30